### ISSUE TITLE
Path with spaces don't get processed correctly

### ISSUE THREAD

**Issue by JakeBelanger** [eflexsystems/node-samba-client#25]
Wrap the file names and paths with double quotes so that they get interpreted correctly. The following worked for me:
```js
runCommand(cmd, path, destination) {
    let workingDir = p.dirname(path);
    let fileName = p.basename(path).replace(singleSlash, '\\');
    let cmdArgs = util.format('%s %s', `"${fileName}"`, `"${destination}"`);

    return this.execute(cmd, cmdArgs, workingDir);
}
```

**Comment by nickperkinslondon** [comment]
Yes please.  This is absolutely necessary.  Lots of windows filenames have spaces in them -- it's very common!  I can make it work by adding my own quotes, but it should happen automatically.

**Comment by jakesjews** [comment]
I'll definitely accept a PR for this if anyone wants to try it out

**Comment by jakesjews** [comment]
@nickperkinslondon can you see if version 4 fixes your issue?

**Comment by JakeBelanger** [comment]
It works, but not it brings up the issue of having the escape single and double quotes in file names. I haven't figured out how to do that properly, so for my use case, I just clean the file names before sending them. However, I do think a fully fledged samba client should be able to send those files without a problem.

Thanks for your work!
